pymc_marketing/clv/models/beta_geo.py

```diff
@@ -101,44 +90,86 @@ class BetaGeoModel(CLVModel):
-        recency: Union[np.ndarray, pd.Series, TensorVariable],
-        T: Union[np.ndarray, pd.Series, TensorVariable],
-        a_prior: Optional[TensorVariable] = None,
-        b_prior: Optional[TensorVariable] = None,
-        alpha_prior: Optional[TensorVariable] = None,
-        r_prior: Optional[TensorVariable] = None,
+        data: pd.DataFrame,
+        model_config: Optional[Dict] = None,
+        sampler_config: Optional[Dict] = None,
     ):
-        super().__init__()
-
-        self.customer_id = customer_id
-        self.frequency = frequency
-        self.recency = recency
-        self.T = T
-
-        a_prior, b_prior, alpha_prior, r_prior = self._process_priors(
-            a_prior, b_prior, alpha_prior, r_prior
+        try:
+            self.customer_id = data["customer_id"]
+        except KeyError:
+            raise KeyError("customer_id column is missing from data")
+        try:
+            self.frequency = data["frequency"]
+        except KeyError:
+            raise KeyError("frequency column is missing from data")
+        try:
+            self.recency = data["recency"]
+        except KeyError:
+            raise KeyError("recency column is missing from data")
+        try:
+            self.T = data["T"]
+        except KeyError:
+            raise KeyError("T column is missing from data")
+        super().__init__(
+            model_config=model_config,
+            sampler_config=sampler_config,
         )
-
+        self.data = data
+        self.a_prior = self.create_distribution_from_prior(
+            self.model_config["a_prior"]["dist"],
+            **self.model_config["a_prior"]["kwargs"],
+        )
+        self.b_prior = self.create_distribution_from_prior(
+            self.model_config["b_prior"]["dist"],
+            **self.model_config["b_prior"]["kwargs"],
+        )
+        self.alpha_prior = self.create_distribution_from_prior(
+            self.model_config["alpha_prior"]["dist"],
+            **self.model_config["alpha_prior"]["kwargs"],
+        )
+        self.r_prior = self.create_distribution_from_prior(
+            self.model_config["r_prior"]["dist"],
+            **self.model_config["r_prior"]["kwargs"],
+        )
+        self._process_priors(self.a_prior, self.b_prior, self.alpha_prior, self.r_prior)
         # each customer's information should be encapsulated by a single data entry
-        if len(np.unique(customer_id)) != len(customer_id):
+        if len(np.unique(self.customer_id)) != len(self.customer_id):
             raise ValueError(
                 "The BetaGeoModel expects exactly one entry per customer. More than"
                 " one entry is currently provided per customer id."
             )
+        self.coords = {"customer_id": self.customer_id}
+
+    @property
+    def default_model_config(self) -> Dict[str, Dict]:
+        return {
+            "a_prior": {"dist": "halfflat", "kwargs": {}},
```

Comment:
We should use the same capitalization convention as pymc distributions

Reply:
true, I realised that as I was already halfway through the implementation. I've decided to leave it to 'aftercare' as I was sure that it won't be the only thing needing some tweaks, and this way I could deliver it faster